code/mob/living/critter/brullbar.dm

```diff
@@ -36,26 +72,192 @@
+					step_away(src, H, 15)
+					src.set_dir(get_dir(src, H))
+					continue
+			. += C
+
+		if (length(.) && prob(10))
+			playsound(src.loc, 'sound/voice/animal/brullbar_roar.ogg', 75, 1)
+			src.visible_message("<span class='alert'><B>[src]</B> roars!</span>")
+
+	seek_scavenge_target(var/range = 9)
+		. = list()
+		for (var/mob/living/M in view(range, get_turf(src)))
+			if (!isdead(M)) continue // eat everything yum
+			if (ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if (H.decomp_stage <= 3 && !H.bioHolder?.HasEffect("husk")) //is dead, isn't a skeleton, isn't a grody husk
+					. += M
+
+	critter_attack(var/mob/target)
+		var/datum/targetable/critter/frenzy = src.abilityHolder.getAbility(/datum/targetable/critter/frenzy)
+		var/datum/targetable/critter/tackle = src.abilityHolder.getAbility(/datum/targetable/critter/tackle)
+		if (!tackle.disabled && tackle.cooldowncheck() && prob(20) && !is_incapacitated(target))
+			tackle.handleCast(target) // no return to wack people with the frenzy after the tackle sometimes
+		if (!frenzy.disabled && frenzy.cooldowncheck() && prob(40))
+			frenzy.handleCast(target)
+			return
+		else if (issilicon(target) && !ON_COOLDOWN(src, "brullbar_messup_silicon", 30 SECONDS))
+			fuck_up_silicons(target)
+			return
+		else
+			return ..()
+
+	critter_scavenge(var/mob/target)
+		if (prob(30))
+			src.visible_message("<span class='alert'><b>[src] devours [target]! Holy shit!</b></span>")
+			playsound(src.loc, 'sound/impact_sounds/Flesh_Break_1.ogg', 50, 1)
+			if (ishuman(target)) new /obj/decal/fakeobjects/skeleton(target.loc)
+			target.ghostize()
+			target.gib()
+			return
+		else
+			src.visible_message("<span class='alert'<b>[src] bites a chunk out of [target]!</b></span>")
+			playsound(src.loc, 'sound/items/eatfood.ogg', 30, 1)
+			for(var/damage_type in src.healthlist)
+				var/datum/healthHolder/hh = src.healthlist[damage_type]
+				hh.HealDamage(10)
+			return
+
+	can_critter_attack()
+		var/datum/targetable/critter/frenzy = src.abilityHolder.getAbility(/datum/targetable/critter/frenzy)
+		return can_act(src,TRUE) && !frenzy.disabled
+
+	proc/retaliate(var/mob/living/attacker) // somewhat stolen from sawfly behaviour, no beating on a confused brullbar
+		var/datum/targetable/critter/tackle = src.abilityHolder.getAbility(/datum/targetable/critter/tackle)
+		if (!istype(attacker, /mob/living/critter/brullbar) || (attacker.health < 0))
+			if (prob(50) && !tackle.disabled && tackle.cooldowncheck() && !isdead(src))
+				src.visible_message("<span class='alert'><b>[src] lunges at [attacker]!</b></span>")
+				playsound(src.loc, 'sound/voice/animal/brullbar_roar.ogg', 50, 1)
+				tackle.handleCast(attacker)
+				ai.interrupt()
+
+	proc/fuck_up_silicons(var/mob/living/silicon/silicon) // taken from orginal object critter behaviour scream
+		if (!isrobot(silicon))
+			src.visible_message("<span class='alert'><B>[src]</B> sniffs at [silicon.name].</span>")
+			sleep(1.5 SECONDS)
```

Comment:
oh no, not `sleep()`. That will hang the entire mob AI loop for 1.5 seconds. `SPAWN()` could work in a pinch, but it's likely better to find a different way to do this.

Reply:
Okay, this is still a little gross, and if you can think of a better way of doing it you should, but #13158 means it's okay.

Reply:
Ive removed the sleeps, it was just copied as is from obj and needs changes anyway